File: go/listnode/listnode.go
package listnode

import (
	"strconv"
)

type ListNode struct {
	Val  int
	Next *ListNode
}

func (l *ListNode) toString() string {
	result := ""

	tmp := l
	for tmp != nil {
		result = result + "->" + strconv.Itoa(tmp.Val)
		tmp = tmp.Next
	}
	return result
}

func getListNode() *ListNode {
	result := ListNode{Val: 1, Next: nil}
	result.Next = &ListNode{2, nil}
	result.Next.Next = &ListNode{3, nil}
	result.Next.Next.Next = &ListNode{4, nil}
	result.Next.Next.Next.Next = &ListNode{5, nil}
	return &result
}

func removeNthFromEnd(head *ListNode, n int) *ListNode {
	result := &ListNode{Next: head, Val: -1}
	slow, fast := result, head
	for n > 0 {
		fast = fast.Next
		n--
	}
	for fast != nil {
		slow = slow.Next
		fast = fast.Next
	}
	slow.Next = slow.Next.Next
	return result.Next
}

File: java/src/main/java/listnode/ListNode.java
package listnode;

/**
 * @author JackCheung
 */
public class ListNode {

    protected int val;

    protected ListNode next;

    public ListNode(int val, ListNode next) {
        this.val = val;
        this.next = next;
    }

    public ListNode(int val) {
        this.val = val;
    }

    public ListNode() {
    }

    public static ListNode build() {
        ListNode result = new ListNode(1);
        result.next = new ListNode(2);
        result.next.next = new ListNode(3);
        result.next.next.next = new ListNode(4);
        result.next.next.next.next = new ListNode(5);
        return result;
    }
}
